File: Dibujoasteriscos2.py
#incompleto
lado = 4 #readline("Lado: ")

for i in range(lado + 1):
    for j in range(10): 
        if j < lado - i:
            print(" ", end = '')
        elif j >= lado - i and j <= 7 + i: 
            print("*", end = '') 
        else:
            print(" ", end = '')

    print("K") 


for i in range(3, 0):
    for j in range(1, 8 + 1):
        if j < lado - i:
            print(" ", end = '')
        elif j >= lado - i and  j <= 7 + i:
            print("*", end = '') 
        else:
            print(" ", end = '')
            
    print("K") 

File: Dibujoasteriscos2.php
<?php
//dibujo de exagono
//incompleto
$lado = readline("Lado: ");

for($i = 0; $i < $lado; $i++){
    
    for($j = 1; $j <= 10 ; $j++){ // $j = $j + 1
        
        if($j < $lado - $i){
            echo " ";
        }
        elseif($j >= $lado - $i && $j <= 7 + $i){
            echo "*";
        }
        else{
            echo " ";
        }
    }
    echo "K\n";
}

for($i = $lado - 1; $i > 0; $i--){
    
    for($j = 1; $j <= 8 ; $j++){ // $j = $j + 1
        
        if($j < $lado - $i){
            echo " ";
        }
        elseif($j >= $lado - $i && $j <=  + $i){
            echo "*";
        }
        else{
            echo " ";
        }
    }
    echo "K\n";
}

